### DISCUSSION TITLE
Player Customization

### DISCUSSION THREAD

**Discussion by HippasusTwo** [Tuxemon/Tuxemon#2279] [Show and tell]
I made some additional customization beyond boy/girl. In particular, the game was really lacking in black character sprites, so I made a male one from scratch and made a female adventurer back sprite. Playtested for ten minutes different options, it seems to work fine.
I also added a penguin, which might be a bit too silly? I can remove that one if so.
https://github.com/HippasusTwo/Tuxemon/tree/enby

**Comment by ultidonki** [comment]
I've modified the eyes and mouth a little bit for these 2 sprites, as I prefer it this way :smiley: No attribution needed if you use it.
I think I added those penguin walking sprites :laughing: I always intended to let people walk around as a penguin, but you can just change the character in the tuxemon.cfg so I left it at that. I guess at some point letting people play as any of the walkable characters might be fun without editing tuxemon.cfg but I guess this is a good way to do it for now. 

I don't have any strong feelings either way whether the penguin is too silly or not so whatever everyone else thinks (as I say, you can still play as it, just by editing the tuxemon.cfg). I also have fond memories of playing as the rock, as well? I don't think there's a walking cycle, you just slide around. I might never have submitted that as a pull request, though. Good times! :smile: 
![adventurerblack2](https://github.com/Tuxemon/Tuxemon/assets/89098769/4470da4b-956d-4336-b47a-bcfac3b8aadb)
![heroineblack2](https://github.com/Tuxemon/Tuxemon/assets/89098769/ec7e604f-2f0a-4b61-83a3-b74631581966)

**Reply by HippasusTwo** [reply]
I'm glad I'm not the only one who likes the idea of walking around as a penguin! There was a single pixel off on the penguin walk cycle that I fixed. And, yeah, this is aimed at casual players: of course anyone who can do even basic editing can set their character to anything they want, but sometimes you just want to be lazy, y'know?

I'm fine with those changes to the two sprites, it's pretty tiny.

**Comment by Sanglorian** [comment]
Thanks @HippasusTwo , that sounds great. Good to have a diversity of options for players.

**Comment by HippasusTwo** [comment]
@Qiangong2 
To address what you said in the pull here: https://github.com/Tuxemon/Tuxemon/pull/2281

The issue of being able to set gender should already be handled, as there is still a gender var which should be set there! So the black character male and the white character male should both have male set in the gender var as gender_male, and same for the female characters.
The dialogue first sets 'race_choice', which then triggers another script that sets the gender in 'debug.yaml'. I added two other gender options, gender_enby and gender_whatever, but that could be compressed down to just one so we just have three pronouns to worry about: He/she/they. Although it would be fairly amusing to have gender_whatever randomly load a different pronoun each time (or 'it'), that would be extra effort so I don't imagine too many people would bother with it for their campaign and would just use the normal three. They is a bit weird as a singular but it has been around for a few hundred years so it's not that weird.

The issue of being able to over-ride and have fewer options is also handled, as I basically copy-pasted the same code for both campaigns. For a new campaign with fewer options (say you wanted them to only play a specific person, like a blonde girl) you would simply use different code.

I think I'll make a penguin suit sprite, to make it clear the person is not actually a penguin but just dressed as them, lol.
(edit: I did.
![penguin_back](https://github.com/Tuxemon/Tuxemon/assets/155398078/e242fe0b-fb51-46a4-a41a-5989a67d432e)

)

The magician, I did notice that the game used it but also that it doesn't quite match the actual walking sprite? Like one of them is hooded and one of them isn't. So I could work on that for you. I have no objections to having it replaced with a different enby character though either. Ideally at some point we'll have a load of character customization in terms of clothing, color, and body build (though some campaigns might want to ban you from playing an old person if you're supposed to be 12).

I noticed @JaskRendix did some work on adding more sprites, particularly clothing color choices it seems?
Sorry about breaking the swimming by the way! I didn't realize. XD

I did notice a weird issue where if I tried to put two dialogue menus one after the other, it bugged up, possibly I did something wrong (need to change background maybe before starting up a new menu???). That's the biggest reason I didn't have race and gender as two different questions for now.

**Comment by Davidliang11292** [comment]
Yes, that's very interesting, but I think we might need to add some Asians?

**Reply by HippasusTwo** [reply]
If you look how sprite art from Japan usually handles Asians and white folks, they often use the same art but just give black hair to the Asian folks. So that shouldn't be too hard.
edit: Alright, Imma replace the disciple with an ambiguously gendered Asian person, since the disciple-knight needed replacing anyway. At some point hopefully we'll have more customization.

**Reply by Davidliang11292** [reply]
OK, Thanks!
